src/targets/gpu/fuse_ck.cpp

```diff
@@ -161,10 +203,81 @@ struct find_ck_gemm
     }
 };
 
+static bool is_mul_module(const module& m)
+{
+    std::vector<std::string> result;
+    for(auto& ins : m)
+    {
+        if(starts_with(ins.name(), "@"))
+            continue;
+        if(contains({"multibroadcast", "contiguous"}, ins.name()))
+            continue;
+        if(ins.name() == "pointwise")
+        {
+            return is_mul_module(*ins.module_inputs().front());
+        }
+        else if(ins.name() == "mul")
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+struct find_ck_gemm_softmax_gemm
+{
+    auto matcher() const
+    {
+        auto gemm1 =
+            match::skip(match::name("contiguous"))(match::name("dot")(is_ck_gemm().bind("gemm1")));
+        auto mul = match::name("pointwise")(match::any_of[match::inputs()](gemm1)).bind("scale");
```

Comment:
This should be `match::either_arg(0, 1)(match::constant().bind("scale"), gemm1).bind("mul")`.